verification/curator-service/ui/src/components/BulkCaseForm.tsx

```diff
@@ -53,54 +92,120 @@ class BulkCaseForm extends React.Component<
             statusMessage: '',
         };
     }
-    // Using a generic type for now; will define case record later.
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    async uploadData(results: ParseResult<Record<string, any>>): Promise<void> {
+
+    createEvents(c: ParsedCase): Event[] {
+        const events = [];
+        // TODO: Add ParsedCase validation.
+        if (c.dateConfirmed) {
+            events.push({
+                name: 'confirmed',
+                dateRange: {
+                    start: c.dateConfirmed,
+                    end: c.dateConfirmed,
+                },
+            });
+        }
+        if (c.hospitalized) {
+            events.push({
+                name: 'hospitalized',
+                dateRange: c.dateHospitalized
+                    ? {
+                        start: c.dateHospitalized,
+                        end: c.dateHospitalized,
+                    }
+                    : undefined,
+            });
+        }
+        if (c.outcome) {
+            events.push({
+                name: 'outcome',
+                dateRange: {
+                    start: c.dateOutcome,
+                    end: c.dateOutcome,
+                },
+                value: c.outcome,
+            });
+        }
+        return events;
+    }
+
+    createGeoResolution(c: ParsedCase): string {
+        if (c.admin3) {
+            return 'Admin3';
+        } else if (c.admin2) {
+            return 'Admin2';
+        } else if (c.admin1) {
+            return 'Admin1';
+        } else {
+            return 'Country';
+        }
+    }
+
+    createLocationQuery(c: ParsedCase): string {
+        return [c.admin3, c.admin2, c.admin1, c.country]
+            .filter((field) => field)
```

Comment:
does this filter do anything? does it filter out null/undefined somehow?

Reply:
Yep:

![image](https://user-images.githubusercontent.com/63060924/86276838-a45f6500-bba3-11ea-9999-66bd7b469925.png)

If there's a cleaner way to do this, LMK!